### PR TITLE
Implement --strict-version-order for migrations

### PR DESCRIPTION
**Pull request by phunehehe**
I have not actually been able to use the option from the command line to see if it works, since I am using v2.2.10 in my project (which is the latest release, but has drifted quite far away from master).

The tests seem to work fine though. Is this something useful?

https://github.com/elixir-ecto/ecto/issues/2614

### CODE REVIEW

File: lib/mix/tasks/ecto.migrate.ex [hunk 1]
```diff
@@ -65,7 +65,8 @@ defmodule Mix.Tasks.Ecto.Migrate do
 
     {opts, _, _} = OptionParser.parse args,
       switches: [all: :boolean, step: :integer, to: :integer, quiet: :boolean,
-                 prefix: :string, pool_size: :integer, log_sql: :boolean],
+                 prefix: :string, pool_size: :integer, log_sql: :boolean,
+                 strict_version_order: :boolean],
```
Comment (josevalim) [inline lib/mix/tasks/ecto.migrate.ex]: Can you please update the docs for this module to mention the --struct-version-order switch?

File: lib/ecto/migrator.ex [hunk 2]
```diff
@@ -93,6 +93,10 @@ defmodule Ecto.Migrator do
           If this can be an issue, we recommend to rollback #{version} and change \
           it to a version later than #{latest}.
           """
+
+          if opts[:strict_version_order] do
+            raise Ecto.MigrationError
+          end
```
Comment (josevalim) [inline lib/ecto/migrator.ex]: Can you please update the docs for this function to mention the `:strict_version_order` option?

Comment (josevalim) [reply lib/ecto/migrator.ex]: Also, what do you think of including the message in the exception? Something like this:

```elixir
message = """
...
"""

if opts[:strict_version_order] do
  raise Ecto.MigrationError, message
else
  Logger.warn(message)
end
```

Comment (phunehehe) [reply lib/ecto/migrator.ex]: Nice idea. Updated.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by wojtekmach** [comment]
> I have not actually been able to use the option from the command line to see if it works, since I am using v2.2.10 in my project

A quick way to test changes is using the [`examples/friends`](https://github.com/elixir-ecto/ecto/tree/master/examples/friends) app, that's what I usually do :)

**Pull request merged**